### PR TITLE
Switch to using new book management API

### PR DESCRIPTION
**Pull request by bojanbg**
Does away with using Django views for authentication and book management. With this change only DRF views are used by `diderot-cli`.

I've manually tested all book management commands except `upload_book` (for which I don't have good test data). I've added some automated tests.

Note that this mustn't be merged before https://github.com/umutacar/lambdabricks/pull/4208, otherwise it will break the cli.

### CODE REVIEW

File: api_calls.py [hunk 1]
```diff
@@ -17,50 +17,40 @@
...
+        self.token_header = {}
         self.client = requests.session()
-        main_page = self.client.get(self.url)
-        code = main_page.status_code
-        if not code == 200:
-            raise err_for_code(code)
-        self.csrftoken = self.client.cookies["csrftoken"]
 
-    # login attempts to log into Diderot using the provided credentials.
+    # login logs into Diderot to get the authentication token
```
[the remark this answers is not in the document]
Comment (rohany) [reply api_calls.py:31]: bump

File: api_calls.py [hunk 2]
```diff
@@ -17,50 +17,40 @@
...
     def login(self, username, password):
         login_data = {
             "username": username,
             "password": password,
-            "csrfmiddlewaretoken": self.csrftoken,
-            "next": "/courses/",
         }
-        login_url = urllib.parse.urljoin(self.url, "login/login/?next=/courses/")
+        login_url = urllib.parse.urljoin(self.url, "/frontend-api/users/login/")
```
Comment (rohany) [inline api_calls.py:37]: mind making this string a constant?

**Review by rohany — CHANGES_REQUESTED** [review]
small requests. I can pick them up, or you could do them on your own time if you'd like. Does the integration test suite in the diderot repo pass with these changes?

**Comment by GodMode-On** [comment]
@umutacar Can we merge it, or I can continue on this branch?

**Comment by umutacar** [comment]
@GodMode-On thank you for the updates.
Let's wait to merge this until December 22 (the end of semester) so that we don't cause things to break.  
As I understand it, we would then need to update the submodule in the Diderot repo, to test this thoroughly.

**Comment by rohany** [comment]
> As I understand it, we would then need to update the submodule in the Diderot repo, to test this thoroughly.

That's correct. It's a good idea to check out this commit in the submodule and run the Diderot tests before merging though, and fixing the bugs that it finds.

**Comment by rohany** [comment]
@moridyn ping me when this is ready for a review

**Comment by moridyn** [comment]
@rohany it is ready for review.

File: api_calls.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rohany) [reply api_calls.py:23]: please fix this

File: api_calls.py [hunk 3]
```diff
@@ -196,33 +177,36 @@ def release_unrelease_chapter(self, course_label, book_label, args, release=True
...
-            release_params["kind"] = "unrelease"
-        self.client.post(MANAGE_BOOK_API.format(course.pk, book.pk), data=release_params)
+        route_params = {
+            "course_id": course.pk,
+            "book_id": book.pk,
+            "chapter_id": chapter.pk,
+            "action": "publish" if release else "retract",
+        }
+        self.client.post(
+            (MANAGE_BOOK_API + "manage-chapters/{chapter_id}/{action}/").format(**route_params)
```
Comment (rohany) [inline api_calls.py:187]: can this be a constant

File: api_calls.py [hunk 4]
```diff
@@ -252,7 +236,18 @@ def upload_chapter(self, course_label, book_label, number, label, args, sleep_ti
...
-            self.client.post(MANAGE_BOOK_API.format(course.pk, book.pk), files=opened_files, data=update_params)
+
+            route_params = {
+                "course_id": course.pk,
+                "book_id": book.pk,
+                "chapter_id": chapter.pk,
+                "action": "content_upload"
+            }
+            self.client.post(
+                (MANAGE_BOOK_API + "manage-chapters/{chapter_id}/{action}/").format(**route_params),
```
Comment (rohany) [inline api_calls.py:247]: same here

File: test_book_management.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    chapter = mk_chapter("chapter", 1)
+    assert not chapter.is_released
+
+    # And an instructor in the course
+    api_client.force_authenticate(pi)
+
+    # When an instructor tries to release the chapter
+    route_params = {"course_id": chapter.course.pk, "book_id": chapter.book.pk,
+                    "chapter_id": chapter.id, "action": "publish"}
+    url = (MANAGE_BOOK_API + "manage-chapters/{chapter_id}/{action}/").format(**route_params)
```
Comment (rohany) [inline test_book_management.py:97]: can use that constant here

**Review by rohany — CHANGES_REQUESTED** [review]
Looks good, mostly nits.

**Review by rohany — COMMENTED** [review]
Looks good! Please squash the commits into a single commit and then merge

**Review by rohany — APPROVED** [review]

**Pull request merged**